tests/test_requests.py

```diff
@@ -2127,6 +2133,7 @@ def test_preparing_url(self, url, expected):
             b"http://*",
             u"http://*.google.com",
             u"http://*",
+            u"http://☃.net/"
         )
     )
     def test_preparing_bad_url(self, url):
```

Comment:
Yeah, this is about preventing wildcards.

To find this kind of thing we really may need to add a custom validator that pulls the hostname apart and checks for wildcards. It's probably sufficient to confirm the URI doesn't start with `u'*.'`.